ADR/ADR004-store-form-schemas-in-an-api.md

```diff
@@ -0,0 +1,144 @@
+# ADR004: Store form schemas in an API
+
+Date: 2022-02-18
+
+## Status
+
+Proposed
+
+## Context
+
+- Manager/Designer components don't need to be as scalable as the runner
+- Runner needs to scale to number of users
+  - Sustainability, runner not needed during quiet times
+- "Published forms" need to be accessible to the runner
+
+### Option 1 - Forms API
+
+```mermaid
+graph TD
+  Manager[Form manager]
+
+  subgraph formsApiService[Forms API]
+    formsApi[API]
+    formStore[(Form datastore)]
+  end
+
+  runnerRenderer[Forms renderer]
+  Manager --> formsApi
+  formsApi --> formStore
+  formStore --> formsApi
+  runnerRenderer --> formsApi
+```
+
+- Stores the forms in a datastore with the API being a separate application
+- Runner depends on API
+- Three applications to deploy with dependencies, too complex?
+- Could cache API responses for speed
+
+### Option 2a - Runner as an API
```

Comment:
A concern with these two (2a/2b) i think would be around the potential for the publishing/unpublishing workflow to end up causing a mis-match if messages get lost on the way to publish/unpublish. Causing the datastore for the manager to be out of sync with what's in the runner. 

You could mitigate this with by doing lookups on the runners published API, but noting the complication